turbopack/crates/turbopack-ecmascript/src/tree_shake/asset.rs

```diff
@@ -342,6 +342,13 @@ impl Module for EcmascriptModulePartAsset {
                 .await?,
         );
 
+        // If the part has no external references, we can skip the analysis
+        if let Some(metadata) = metadata.get(&part_id) {
+            if !metadata.have_external_references {
+                return Ok(Vc::cell(references));
+            }
+        }
```

Comment:
The analysis does more than extracting ESM references. It also analyses `require(...)` and `process.env.SOMETHING`.